# THE BATTLE OF NEIGHBORHOODS

### 1) A description of the problem and a discussion of the background.

This project has the objective to explorer the best location to have a Asian restaurant throughout the Bronx of New York. New York is a metropolitan area with more than 11 million people living and is the largest city in the United States, New York contains the highest total Asian population of any U.S. city.
According to the 2010 census, 6.0% of New York City was of Chinese ethnicity, with about eighty percent of Chinese New Yorkers living in the boroughs of Queens and Brooklyn alone; New York City itself contains by far the highest ethnic Chinese population of any individual city outside Asia, estimated at 628,763 as of 2017.People of Korean heritage made up 1.2% of the city's population, and Japanese or Japanese American heritage 0.3%. Filipino and Filipino Americans were the largest southeast Asian ethnic group at 0.8%, followed by people of Vietnamese heritage, who made up 0.2% of New York City's population in 2010. Indian and Indian Americans comprise the largest South Asian group, comprising 2.4% of the city's population, with Bangladeshi and Bangladeshi Americans and people of Pakistani heritage at 0.7% and 0.5%, respectively.
With its diverse culture, comes diverse food items. There are many restaurants in New York City, each belonging to different categories like Chinese, Japanese, Thailand.

### 2) The problem 

The objective is to determine the frequency of the Asian restaurants in the neighborhoods of Bronx to visualize the degree of competition that can face in differents places

### 3) The code

In [226]:
# IMPORT
import numpy as np # library to handle data in a vectorized manner

import pandas as pd # library for data analsysis
pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)

import json # library to handle JSON files

!conda install -c conda-forge geopy --yes # uncomment this line if you haven't completed the Foursquare API lab
from geopy.geocoders import Nominatim # convert an address into latitude and longitude values

import requests # library to handle requests
from pandas.io.json import json_normalize # tranform JSON file into a pandas dataframe

# Matplotlib and associated plotting modules
import matplotlib.cm as cm
import matplotlib.colors as colors

# import k-means from clustering stage
from sklearn.cluster import KMeans

!conda install -c conda-forge folium=0.5.0 --yes # uncomment this line if you haven't completed the Foursquare API lab
import folium # map rendering library

print('Libraries imported.')

usage: conda-script.py [-h] [-V] command ...
conda-script.py: error: unrecognized arguments: # uncomment this line if you haven't completed the Foursquare API lab


Libraries imported.


usage: conda-script.py [-h] [-V] command ...
conda-script.py: error: unrecognized arguments: # uncomment this line if you haven't completed the Foursquare API lab


In [227]:
import json
import requests
ny=requests.get('https://cocl.us/new_york_dataset')
print('Data downloaded!')

Data downloaded!


In [228]:
newyork_data = json.loads(ny.text)
neighborhoods_data = newyork_data['features']
neighborhoods_data[0]

# define the dataframe columns
column_names = ['Borough', 'Neighborhood', 'Latitude', 'Longitude'] 
# instantiate the dataframe
neighborhoods = pd.DataFrame(columns=column_names)

In [229]:
neighborhoods

Empty DataFrame
Columns: [Borough, Neighborhood, Latitude, Longitude]
Index: []

In [230]:
for data in neighborhoods_data:
    borough = neighborhood_name = data['properties']['borough'] 
    neighborhood_name = data['properties']['name']
        
    neighborhood_latlon = data['geometry']['coordinates']
    neighborhood_lat = neighborhood_latlon[1]
    neighborhood_lon = neighborhood_latlon[0]
    
    neighborhoods = neighborhoods.append({'Borough': borough,
                                          'Neighborhood': neighborhood_name,
                                          'Latitude': neighborhood_lat,
                                          'Longitude': neighborhood_lon}, ignore_index=True)

In [231]:
neighborhoods.head()

Borough Neighborhood   Latitude  Longitude
0   Bronx    Wakefield  40.894705 -73.847201
1   Bronx   Co-op City  40.874294 -73.829939
2   Bronx  Eastchester  40.887556 -73.827806
3   Bronx    Fieldston  40.895437 -73.905643
4   Bronx    Riverdale  40.890834 -73.912585

In [232]:
print('The dataframe has {} boroughs and {} neighborhoods.'.format(
        len(neighborhoods['Borough'].unique()),
        neighborhoods.shape[0]
    )
)

The dataframe has 5 boroughs and 306 neighborhoods.


In [233]:
bronx_data = neighborhoods[neighborhoods['Borough'] == 'Bronx'].reset_index(drop=True)
bronx_data.head()

Borough Neighborhood   Latitude  Longitude
0   Bronx    Wakefield  40.894705 -73.847201
1   Bronx   Co-op City  40.874294 -73.829939
2   Bronx  Eastchester  40.887556 -73.827806
3   Bronx    Fieldston  40.895437 -73.905643
4   Bronx    Riverdale  40.890834 -73.912585

In [234]:
address = 'Bronx, NY'

geolocator = Nominatim(user_agent="ny_explorer")
location = geolocator.geocode(address)
latitude = location.latitude
longitude = location.longitude
print('The geograpical coordinate of Bronx are {}, {}.'.format(latitude, longitude))

The geograpical coordinate of Bronx are 40.8466508, -73.8785937.


# CREATE A MAP OF BRONX CITY

In [235]:
# create map of New York using latitude and longitude values
map_bronx = folium.Map(location=[latitude, longitude], zoom_start=10)

# add markers to map
for lat, lng, borough, neighborhood in zip(neighborhoods['Latitude'], neighborhoods['Longitude'], neighborhoods['Borough'], neighborhoods['Neighborhood']):
    label = '{}, {}'.format(neighborhood, borough)
    label = folium.Popup(label, parse_html=True)
    folium.CircleMarker(
        [lat, lng],
        radius=5,
        popup=label,
        color='blue',
        fill=True,
        fill_color='#3186cc',
        fill_opacity=0.7,
        parse_html=False).add_to(map_bronx)  
    
map_bronx

In [236]:
CLIENT_ID = 'AWT4QK5VTLAHVAMPMKNEJDP3TL25SX25VKAMQWN5UL0LUMYS' 
CLIENT_SECRET = 'JCQYEAAMVKPXNVYVVPWJVLJD1YBRLHVO24SBYDKMTSJKJF2Z' 
VERSION = '20180605' 

print('Your credentails:')
print('CLIENT_ID: ' + CLIENT_ID)
print('CLIENT_SECRET:' + CLIENT_SECRET)

Your credentails:
CLIENT_ID: AWT4QK5VTLAHVAMPMKNEJDP3TL25SX25VKAMQWN5UL0LUMYS
CLIENT_SECRET:JCQYEAAMVKPXNVYVVPWJVLJD1YBRLHVO24SBYDKMTSJKJF2Z


In [237]:
bronx_data.loc[0, 'Neighborhood']

'Wakefield'

In [238]:
neighborhood_latitude = bronx_data.loc[0, 'Latitude'] # neighborhood latitude value
neighborhood_longitude = bronx_data.loc[0, 'Longitude'] # neighborhood longitude value

neighborhood_name = bronx_data.loc[0, 'Neighborhood'] # neighborhood name

print('Latitude and longitude values of {} are {}, {}.'.format(neighborhood_name, 
                                                               neighborhood_latitude, 
                                                               neighborhood_longitude))

Latitude and longitude values of Wakefield are 40.89470517661, -73.84720052054902.


In [239]:
limit=100
radius=500

url = 'https://api.foursquare.com/v2/venues/explore?&client_id={}&client_secret={}&v={}&ll={},{}&radius={}&limit={}'.format(CLIENT_ID,
       CLIENT_SECRET,
       VERSION,
       neighborhood_latitude,
       neighborhood_longitude,
       radius,limit)

In [240]:
results=requests.get(url).json()
results

{'meta': {'code': 200, 'requestId': '5f454a964f3e540acf490bfb'},
 'response': {'suggestedFilters': {'header': 'Tap to show:',
   'filters': [{'name': 'Open now', 'key': 'openNow'}]},
  'headerLocation': 'Wakefield',
  'headerFullLocation': 'Wakefield, Bronx',
  'headerLocationGranularity': 'neighborhood',
  'totalResults': 9,
  'suggestedBounds': {'ne': {'lat': 40.899205181110005,
    'lng': -73.84125857127495},
   'sw': {'lat': 40.89020517211, 'lng': -73.8531424698231}},
  'groups': [{'type': 'Recommended Places',
    'name': 'recommended',
    'items': [{'reasons': {'count': 0,
       'items': [{'summary': 'This spot is popular',
         'type': 'general',
         'reasonName': 'globalInteractionReason'}]},
      'venue': {'id': '4c537892fd2ea593cb077a28',
       'name': 'Lollipops Gelato',
       'location': {'address': '4120 Baychester Ave',
        'crossStreet': 'Edenwald & Bussing Ave',
        'lat': 40.894123150205274,
        'lng': -73.84589162362325,
        'labeledLatLn

In [241]:
# function that extracts the category of the venue
def get_category_type(row):
    try:
        categories_list = row['categories']
    except:
        categories_list = row['venue.categories']
        
    if len(categories_list) == 0:
        return None
    else:
        return categories_list[0]['name']

In [242]:
venues = results['response']['groups'][0]['items']
    
nearby_venues = json_normalize(venues) # flatten JSON

# filter columns
filtered_columns = ['venue.name', 'venue.categories', 'venue.location.lat', 'venue.location.lng']
nearby_venues =nearby_venues.loc[:, filtered_columns]

# filter the category for each row
nearby_venues['venue.categories'] = nearby_venues.apply(get_category_type, axis=1)

# clean columns
nearby_venues.columns = [col.split(".")[-1] for col in nearby_venues.columns]

nearby_venues.head()

<ipython-input-242-561c05f0fdd1>:3: FutureWarning: pandas.io.json.json_normalize is deprecated, use pandas.json_normalize instead
  nearby_venues = json_normalize(venues) # flatten JSON


name      categories        lat        lng
0  Lollipops Gelato    Dessert Shop  40.894123 -73.845892
1          Rite Aid        Pharmacy  40.896649 -73.844846
2  Carvel Ice Cream  Ice Cream Shop  40.890487 -73.848568
3         Walgreens        Pharmacy  40.896528 -73.844700
4             Shell     Gas Station  40.894187 -73.845862

In [243]:
print('{} venues were returned by Foursquare.'.format(nearby_venues.shape[0]))

9 venues were returned by Foursquare.


# EXPLORING NEIGHBORHOOD

In [244]:
def getNearbyVenues(names, latitudes, longitudes, radius=500, LIMIT=100):
    
    venues_list=[]
    for name, lat, lng in zip(names, latitudes, longitudes):
        print(name)
            
        # create the API request URL
        url = 'https://api.foursquare.com/v2/venues/explore?&client_id={}&client_secret={}&v={}&ll={},{}&radius={}&limit={}'.format(
            CLIENT_ID, 
            CLIENT_SECRET, 
            VERSION, 
            lat, 
            lng, 
            radius, 
            LIMIT)
            
        # make the GET request
        results = requests.get(url).json()["response"]['groups'][0]['items']
        
        # return only relevant information for each nearby venue
        venues_list.append([(
            name, 
            lat, 
            lng, 
            v['venue']['name'], 
            v['venue']['location']['lat'], 
            v['venue']['location']['lng'],  
            v['venue']['categories'][0]['name']) for v in results])

    nearby_venues = pd.DataFrame([item for venue_list in venues_list for item in venue_list])
    nearby_venues.columns = ['Neighborhood', 
                  'Neighborhood Latitude', 
                  'Neighborhood Longitude', 
                  'Venue', 
                  'Venue Latitude', 
                  'Venue Longitude', 
                  'Venue Category']
    
    return(nearby_venues)

In [245]:
nearby_venues

name      categories        lat        lng
0  Lollipops Gelato    Dessert Shop  40.894123 -73.845892
1          Rite Aid        Pharmacy  40.896649 -73.844846
2  Carvel Ice Cream  Ice Cream Shop  40.890487 -73.848568
3         Walgreens        Pharmacy  40.896528 -73.844700
4             Shell     Gas Station  40.894187 -73.845862
5           Dunkin'      Donut Shop  40.890459 -73.849089
6            Subway  Sandwich Place  40.890468 -73.849152
7      Central Deli   Deli / Bodega  40.896728 -73.844387
8   Koss Quick Wash      Laundromat  40.891281 -73.849904

In [246]:
bronx_venues = getNearbyVenues(names=bronx_data['Neighborhood'],
                                   latitudes=bronx_data['Latitude'],
                                   longitudes=bronx_data['Longitude'],
                                   )

Wakefield
Co-op City
Eastchester
Fieldston
Riverdale
Kingsbridge
Woodlawn
Norwood
Williamsbridge
Baychester
Pelham Parkway
City Island
Bedford Park
University Heights
Morris Heights
Fordham
East Tremont
West Farms
High  Bridge
Melrose
Mott Haven
Port Morris
Longwood
Hunts Point
Morrisania
Soundview
Clason Point
Throgs Neck
Country Club
Parkchester
Westchester Square
Van Nest
Morris Park
Belmont
Spuyten Duyvil
North Riverdale
Pelham Bay
Schuylerville
Edgewater Park
Castle Hill
Olinville
Pelham Gardens
Concourse
Unionport
Edenwald
Claremont Village
Concourse Village
Mount Eden
Mount Hope
Bronxdale
Allerton
Kingsbridge Heights


In [247]:
print(bronx_venues.shape)
bronx_venues.head()

(1206, 7)


Neighborhood  Neighborhood Latitude  Neighborhood Longitude  \
0    Wakefield              40.894705              -73.847201   
1    Wakefield              40.894705              -73.847201   
2    Wakefield              40.894705              -73.847201   
3    Wakefield              40.894705              -73.847201   
4    Wakefield              40.894705              -73.847201   

              Venue  Venue Latitude  Venue Longitude  Venue Category  
0  Lollipops Gelato       40.894123       -73.845892    Dessert Shop  
1          Rite Aid       40.896649       -73.844846        Pharmacy  
2  Carvel Ice Cream       40.890487       -73.848568  Ice Cream Shop  
3         Walgreens       40.896528       -73.844700        Pharmacy  
4             Shell       40.894187       -73.845862     Gas Station

In [248]:
bronx_venues.groupby('Neighborhood').count()

Neighborhood Latitude  Neighborhood Longitude  Venue  \
Neighborhood                                                                
Allerton                                31                      31     31   
Baychester                              18                      18     18   
Bedford Park                            34                      34     34   
Belmont                                100                     100    100   
Bronxdale                               12                      12     12   
Castle Hill                              7                       7      7   
City Island                             25                      25     25   
Claremont Village                       17                      17     17   
Clason Point                            10                      10     10   
Co-op City                              14                      14     14   
Concourse                               26                      26     26   
Concourse Village                       35                      35     35   
Country Club                             4                       4      4   
East Tremont                            19                      19     19   
Eastchester                             19                      19     19   
Edenwald                                 4                       4      4   
Edgewater Park                          20                      20     20   
Fieldston                                4                       4      4   
Fordham                                 86                      86     86   
High  Bridge                            29                      29     29   
Hunts Point                             13                      13     13   
Kingsbridge                             67                      67     67   
Kingsbridge Heights                     33                      33     33   
Longwood                                 8                       8      8   
Melrose                                 30                      30     30   
Morris Heights                          12                      12     12   
Morris Park                             25                      25     25   
Morrisania                              22                      22     22   
Mott Haven                              18                      18     18   
Mount Eden                              32                      32     32   
Mount Hope                              15                      15     15   
North Riverdale                         22                      22     22   
Norwood                                 33                      33     33   
Olinville                               10                      10     10   
Parkchester                             33                      33     33   
Pelham Bay                              41                      41     41   
Pelham Gardens                          16                      16     16   
Pelham Parkway                          26                      26     26   
Port Morris                             12                      12     12   
Riverdale                               11                      11     11   
Schuylerville                           17                      17     17   
Soundview                               18                      18     18   
Spuyten Duyvil                           6                       6      6   
Throgs Neck                             13                      13     13   
Unionport                               20                      20     20   
University Heights                      22                      22     22   
Van Nest                                19                      19     19   
Wakefield                                9                       9      9   
West Farms                              26                      26     26   
Westchester Square                      31                      31     31   
Williamsbridge   

In [249]:
print('There are {} uniques categories.'.format(len(bronx_venues['Venue Category'].unique())))

There are 179 uniques categories.


# ANALYSE EACH NEIGHBORHOOD

In [307]:
# one hot encoding
bronx_onehot = pd.get_dummies(bronx_venues[['Venue Category']], prefix="", prefix_sep="")

# add neighborhood column back to dataframe
bronx_onehot['Neighborhood'] = bronx_venues['Neighborhood'] 

# move neighborhood column to the first column
fixed_columns = [bronx_onehot.columns[-1]] + list(bronx_onehot.columns[:-1])
bronx_onehot = bronx_onehot[fixed_columns]

bronx_onehot.head()

Neighborhood  Accessories Store  African Restaurant  American Restaurant  \
0    Wakefield                  0                   0                    0   
1    Wakefield                  0                   0                    0   
2    Wakefield                  0                   0                    0   
3    Wakefield                  0                   0                    0   
4    Wakefield                  0                   0                    0   

   Arcade  Arepa Restaurant  Art Gallery  Art Museum  Arts & Crafts Store  \
0       0                 0            0           0                    0   
1       0                 0            0           0                    0   
2       0                 0            0           0                    0   
3       0                 0            0           0                    0   
4       0                 0            0           0                    0   

   Asian Restaurant  Athletics & Sports  Auto Garage  Automotive Shop  \
0                 0                   0            0                0   
1                 0                   0            0                0   
2                 0                   0            0                0   
3                 0                   0            0                0   
4                 0                   0            0                0   

   BBQ Joint  Bagel Shop  Bakery  Bank  Bar  Baseball Field  Basketball Court  \
0          0           0       0     0    0               0                 0   
1          0           0       0     0    0               0                 0   
2          0           0       0     0    0               0                 0   
3          0           0       0     0    0               0                 0   
4          0           0       0     0    0               0                 0   

   Beer Bar  Bike Trail  Board Shop  Boat or Ferry  Bookstore  Bowling Alley  \
0         0           0           0              0          0              0   
1         0           0           0              0          0              0   
2         0           0           0              0          0              0   
3         0           0           0              0          0              0   
4         0           0           0              0          0              0   

   Breakfast Spot  Brewery  Buffet  Building  Burger Joint  Bus Line  \
0               0        0       0         0             0         0   
1               0        0       0         0             0         0   
2               0        0       0         0             0         0   
3               0        0       0         0             0         0   
4               0        0       0         0             0         0   

   Bus Station  Bus Stop  Business Service  Café  Candy Store  \
0            0         0                 0     0            0   
1            0         0                 0     0            0   
2            0         0                 0     0            0   
3            0         0                 0     0            0   
4            0         0                 0     0            0   

   Caribbean Restaurant  Check Cashing Service  Cheese Shop  \
0                     0                      0            0   
1                     0                      0            0   
2                     0                      0            0   
3                     0                      0            0   
4                     0                      0            0   

   Chinese Restaurant  Clothing Store  Coffee Shop  Comfort Food Restaurant  \
0                   0               0            0                        0   
1                   0               0            0                        0   
2                   0               0            0                        0   
3                   0               0            0                        0   
4                   0               0            0               

In [343]:
bronx_grouped = bronx_onehot.groupby('Neighborhood').mean().reset_index()
bronx_grouped

Neighborhood  Accessories Store  African Restaurant  \
0              Allerton           0.000000            0.000000   
1            Baychester           0.000000            0.000000   
2          Bedford Park           0.000000            0.000000   
3               Belmont           0.000000            0.000000   
4             Bronxdale           0.000000            0.000000   
5           Castle Hill           0.000000            0.000000   
6           City Island           0.000000            0.000000   
7     Claremont Village           0.000000            0.000000   
8          Clason Point           0.000000            0.000000   
9            Co-op City           0.000000            0.000000   
10            Concourse           0.000000            0.000000   
11    Concourse Village           0.000000            0.000000   
12         Country Club           0.000000            0.000000   
13         East Tremont           0.000000            0.000000   
14          Eastchester           0.000000            0.000000   
15             Edenwald           0.000000            0.000000   
16       Edgewater Park           0.000000            0.000000   
17            Fieldston           0.000000            0.000000   
18              Fordham           0.011628            0.011628   
19         High  Bridge           0.000000            0.000000   
20          Hunts Point           0.000000            0.000000   
21          Kingsbridge           0.000000            0.000000   
22  Kingsbridge Heights           0.000000            0.000000   
23             Longwood           0.000000            0.000000   
24              Melrose           0.000000            0.000000   
25       Morris Heights           0.000000            0.000000   
26          Morris Park           0.000000            0.000000   
27           Morrisania           0.000000            0.000000   
28           Mott Haven           0.000000            0.000000   
29           Mount Eden           0.000000            0.000000   
30           Mount Hope           0.000000            0.000000   
31      North Riverdale           0.000000            0.000000   
32              Norwood           0.000000            0.000000   
33            Olinville           0.000000            0.000000   
34          Parkchester           0.000000            0.000000   
35           Pelham Bay           0.000000            0.000000   
36       Pelham Gardens           0.000000            0.000000   
37       Pelham Parkway           0.000000            0.000000   
38          Port Morris           0.000000            0.000000   
39            Riverdale           0.000000            0.000000   
40        Schuylerville           0.000000            0.000000   
41            Soundview           0.000000            0.000000   
42       Spuyten Duyvil           0.000000            0.000000   
43          Throgs Neck           0.000000            0.000000   
44            Unionport           0.000000            0.000000   
45   University Heights           0.000000            0.045455   
46             Van Nest           0.000000            0.000000   
47            Wakefield           0.000000            0.000000   
48           West Farms           0.000000            0.000000   
49   Westchester Square           0.000000            0.000000   
50       Williamsbridge           0.000000            0.000000   
51             Woodlawn           0.000000            0.000000   

    American Restaurant    Arcade  Arepa Restaurant  Art Gallery  Art Museum  \
0              0.000000  0.000000              0.00     0.000000    0.000000   
1              0.000000  0.055556              0.00     0.000000    0.000000   
2              0.000000  0.000000              0.00     0.000000    0.000000   
3              0.010000  0.000000              0.00     0.000000    0.000000   
4              0.000000  0.000000              0.00     0.000000    0.000000   
5              0.000000  0.00000

In [344]:
bronx_asian=bronx_grouped.loc[bronx_grouped['Asian Restaurant']>0]
bronx_asian

Neighborhood  Accessories Store  African Restaurant  \
16      Edgewater Park                0.0                 0.0   
19        High  Bridge                0.0                 0.0   
34         Parkchester                0.0                 0.0   
35          Pelham Bay                0.0                 0.0   
43         Throgs Neck                0.0                 0.0   
49  Westchester Square                0.0                 0.0   

    American Restaurant  Arcade  Arepa Restaurant  Art Gallery  Art Museum  \
16             0.050000     0.0               0.0          0.0         0.0   
19             0.000000     0.0               0.0          0.0         0.0   
34             0.060606     0.0               0.0          0.0         0.0   
35             0.000000     0.0               0.0          0.0         0.0   
43             0.076923     0.0               0.0          0.0         0.0   
49             0.000000     0.0               0.0          0.0         0.0   

    Arts & Crafts Store  Asian Restaurant  Athletics & Sports  Auto Garage  \
16                  0.0          0.050000                 0.0          0.0   
19                  0.0          0.034483                 0.0          0.0   
34                  0.0          0.030303                 0.0          0.0   
35                  0.0          0.024390                 0.0          0.0   
43                  0.0          0.076923                 0.0          0.0   
49                  0.0          0.032258                 0.0          0.0   

    Automotive Shop  BBQ Joint  Bagel Shop   Bakery      Bank       Bar  \
16              0.0    0.00000     0.00000  0.00000  0.000000  0.050000   
19              0.0    0.00000     0.00000  0.00000  0.000000  0.000000   
34              0.0    0.00000     0.00000  0.00000  0.030303  0.000000   
35              0.0    0.02439     0.02439  0.02439  0.073171  0.000000   
43              0.0    0.00000     0.00000  0.00000  0.000000  0.076923   
49              0.0    0.00000     0.00000  0.00000  0.032258  0.032258   

    Baseball Field  Basketball Court  Beer Bar  Bike Trail  Board Shop  \
16        0.000000               0.0       0.0         0.0         0.0   
19        0.000000               0.0       0.0         0.0         0.0   
34        0.000000               0.0       0.0         0.0         0.0   
35        0.000000               0.0       0.0         0.0         0.0   
43        0.000000               0.0       0.0         0.0         0.0   
49        0.032258               0.0       0.0         0.0         0.0   

    Boat or Ferry  Bookstore  Bowling Alley  Breakfast Spot  Brewery  Buffet  \
16            0.0        0.0            0.0             0.0      0.0     0.0   
19            0.0        0.0            0.0             0.0      0.0     0.0   
34            0.0        0.0            0.0             0.0      0.0     0.0   
35            0.0        0.0            0.0             0.0      0.0     0.0   
43            0.0        0.0            0.0             0.0      0.0     0.0   
49            0.0        0.0            0.0             0.0      0.0     0.0   

    Building  Burger Joint  Bus Line  Bus Station  Bus Stop  Business Service  \
16  0.000000           0.0       0.0     0.000000       0.0               0.0   
19  0.000000           0.0       0.0     0.068966       0.0               0.0   
34  0.000000           0.0       0.0     0.000000       0.0               0.0   
35  0.000000           0.0       0.0     0.000000       0.0               0.0   
43  0.000000           0.0       0.0     0.000000       0.0               0.0   
49  0.032258           0.0       0.0     0.000000       0.0               0.0   

    Café  Candy Store  Caribbean Restaurant  Check Cashing Service  \
16   0.0          0.0              0.000000               0.000000   
19   0.0          0.0              0.000000               0.034483   
34   0.0          0.0              0.030303               0.000000   
35   0

In [345]:
bronx_grouped.shape

(52, 180)

In [346]:
num_top_venues = 5

for hood in bronx_asian['Neighborhood']:
    print("----"+hood+"----")
    temp = bronx_asian[bronx_asian['Neighborhood'] == hood].T.reset_index()
    temp.columns = ['venue','freq']
    temp = temp.iloc[1:]
    temp['freq'] = temp['freq'].astype(float)
    temp = temp.round({'freq': 2})
    print(temp.sort_values('freq', ascending=False).reset_index(drop=True).head(num_top_venues))
    print('\n')

----Edgewater Park----
                venue  freq
0  Italian Restaurant  0.20
1         Pizza Place  0.10
2       Deli / Bodega  0.10
3         Coffee Shop  0.10
4          Donut Shop  0.05


----High  Bridge----
                       venue  freq
0         Chinese Restaurant  0.17
1                Pizza Place  0.10
2                Bus Station  0.07
3              Deli / Bodega  0.07
4  Latin American Restaurant  0.07


----Parkchester----
                 venue  freq
0          Supermarket  0.12
1          Pizza Place  0.09
2   Chinese Restaurant  0.06
3           Kids Store  0.06
4  American Restaurant  0.06


----Pelham Bay----
                  venue  freq
0    Italian Restaurant  0.10
1                  Bank  0.07
2  Fast Food Restaurant  0.05
3            Donut Shop  0.05
4        Sandwich Place  0.05


----Throgs Neck----
               venue  freq
0      Deli / Bodega  0.15
1  Mobile Phone Shop  0.08
2   Asian Restaurant  0.08
3        Pizza Place  0.08
4                Bar  

In [369]:
bronx_asian['Asian Restaurant']

16    0.050000
19    0.034483
34    0.030303
35    0.024390
43    0.076923
49    0.032258
Name: Asian Restaurant, dtype: float64

According to the table that was generated, the Neighborhoods that can face competition to create a asian restaurant are:

- Throgs Neck: with freq = 0.076923
- Edgewater Park: with freq = 0.05000
- High Bridge: with freq = 0.34483
- Westchester Square: with freq = 0.032258
- Parkchester: with freq = 0.030303
- Pelham Bay: with freq = 0.024390

# Analyze other services

In [357]:
def return_most_common_venues(row, num_top_venues):
    row_categories = row.iloc[1:]
    row_categories_sorted = row_categories.sort_values(ascending=False)
    
    return row_categories_sorted.index.values[0:num_top_venues]

In [364]:
num_top_venues = 10

indicators = ['st', 'nd', 'rd']

# create columns according to number of top venues
columns = ['Neighborhood']
for ind in np.arange(num_top_venues):
    try:
        columns.append('{}{} Most Common Venue'.format(ind+1, indicators[ind]))
    except:
        columns.append('{}th Most Common Venue'.format(ind+1))

# create a new dataframe
neighborhoods_venues_sorted = pd.DataFrame(columns=columns)
neighborhoods_venues_sorted['Neighborhood'] = bronx_asian['Neighborhood']

for ind in np.arange(bronx_asian.shape[0]):
    neighborhoods_venues_sorted.iloc[ind, 1:] = return_most_common_venues(bronx_asian.iloc[ind, :], num_top_venues)

neighborhoods_venues_sorted.head(6)

Neighborhood 1st Most Common Venue 2nd Most Common Venue  \
16      Edgewater Park    Italian Restaurant           Pizza Place   
19        High  Bridge    Chinese Restaurant           Pizza Place   
34         Parkchester           Supermarket           Pizza Place   
35          Pelham Bay    Italian Restaurant                  Bank   
43         Throgs Neck         Deli / Bodega           Coffee Shop   
49  Westchester Square  Fast Food Restaurant              Pharmacy   

        3rd Most Common Venue 4th Most Common Venue 5th Most Common Venue  \
16              Deli / Bodega           Coffee Shop                   Bar   
19  Latin American Restaurant              Pharmacy         Deli / Bodega   
34        American Restaurant            Kids Store    Chinese Restaurant   
35             Sandwich Place        Cosmetics Shop     Convenience Store   
43                 Sports Bar          Liquor Store             Juice Bar   
49                Pizza Place        Sandwich Place            Donut Shop   

   6th Most Common Venue 7th Most Common Venue       8th Most Common Venue  \
16                   Pub                  Park                  Donut Shop   
19           Bus Station        Ice Cream Shop  Construction & Landscaping   
34         Women's Store        Shipping Store          Mexican Restaurant   
35  Fast Food Restaurant            Donut Shop              Ice Cream Shop   
43    Chinese Restaurant    Italian Restaurant            Asian Restaurant   
49           Supermarket         Metro Station          Mexican Restaurant   

        9th Most Common Venue 10th Most Common Venue  
16                        Spa           Liquor Store  
19                       Park  Check Cashing Service  
34             Mattress Store         Sandwich Place  
35  Latin American Restaurant            Snack Place  
43          Mobile Phone Shop    American Restaurant  
49  Latin American Restaurant     Spanish Restaurant

# CONCLUSION

In this project was possible to identify that Asian Restaurants are not common in the Borough of Bronx, analysing the map is also possible to notice that the competition between asian restaurants is not so relevant, but the neighborhood that the frequency is lowest in Bronx is Pelham Bay ans the highest is Throgs Neck.
But with a complete anlysis in the Data to the neighborhoods which presents the highest values of frequency in Asian Restaurants we identify that none of then has this kind of food as Most common, except High Bridge, maybe is necessary to think in another kind of food forward considering that it is not popular in these regions.